components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -195,26 +201,27 @@ func (sb *sensorBase) Reconfigure(ctx context.Context, deps resource.Dependencie
 
 	if sb.velocities != nil && len(newConf.ControlParameters) != 0 {
 		// assign linear and angular PID correctly based on the given type
-		var linear, angular control.PIDConfig
 		for _, c := range newConf.ControlParameters {
 			switch c.Type {
 			case typeLinVel:
-				linear = c
+				// configPIDVals at index 0 is linear
+				sb.configPIDVals[0] = c
```

Comment:
let's rename c so the meaning of the symbol matches the struct you're populating. You know that the control parameters are pid values, but someone else may not.